compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -804,4 +885,67 @@ private boolean populateRecordFields(BLangExpression expr, BLangConstant constan
         }
         return true;
     }
+
+    private BType createTupleType(BLangExpression expr, BConstantSymbol constantSymbol, Location pos,
+                                  Object constValue, SymbolEnv env) {
+        if (expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+            return expr.getBType();
+        }
+
+        List<BLangConstantValue> constValueList = (List<BLangConstantValue>) constValue;
+
+        List<BType> tupleTypes = new ArrayList<>(constValueList.size());
```

Comment:
shall we move `tupleTypes` initilization after next for loop because we can `return null`. And this situation we dont need to allocate a new array